src/main/flight/interpolated_setpoint.c

```diff
@@ -68,46 +69,56 @@ FAST_CODE_NOINLINE float interpolatedSpApply(int axis, float pidFrequency, bool
 
         // get the number of interpolation steps either dynamically based on RX refresh rate
         // or manually based on ffSpread configuration property
+        uint16_t rcIntervalSteps =  (uint16_t) ((currentRxRefreshRate + 1000) * pidFrequency * 1e-6f + 0.5f);
+
         if (ffSpread) {
-            interpolationSteps[axis] = (uint16_t) ((ffSpread + 1.0f) * 0.001f * pidFrequency);
+            interpolationSteps[axis] =  (uint16_t) ((ffSpread + 1.0f) * 0.001f * pidFrequency);
         } else {
-            interpolationSteps[axis] = (uint16_t) ((currentRxRefreshRate + 1000) * pidFrequency * 1e-6f + 0.5f);
+            interpolationSteps[axis] = rcIntervalSteps;
         }
+        interpolationSteps[axis] = MIN(interpolationSteps[axis], 2 * rcIntervalSteps);
 
         // interpolate stick deflection
+        deflectionChangePerIteration[axis] = deflectionReservoir[axis] / rcIntervalSteps;
         deflectionReservoir[axis] += rawDeflection - prevRawDeflection[axis];
-        deflectionChangePerIteration[axis] = deflectionReservoir[axis] / interpolationSteps[axis];
+        deflectionChangePerIteration[axis] += (rawDeflection - prevRawDeflection[axis]) / interpolationSteps[axis];
```

Comment:
This is exactly what I didn't like on FF PR - the code is too complicated and every change is making it even more cryptic.

Can you explain rationale behind this code? All I see is that it `does work`, but it would take a lot of time to understand it ...

Reply:
Yeah @ledvinap fair critique as always. Here's what I wanted to have happen: the ff from one rc interval gets applied (spread out) over the duration of ff_spread. During that time the contribution from that frame should be constant. What the code did however was to spread it out way longer by applying the spreading operation again on the remainder plus the new ff when the next frame comes in, effectively implementing a lpf which you pointed out.

I think where we are now the code can really be simplified a lot. stay tuned.

Reply:
Well .. the whole _reservoir_ idea is IMO complicating it.
What you want to do is to take delta when it arrives, adapt it to PID update rate (divide by time since last RC update/ multiply by PID time) and use this value for _spread_ time or until next RX update arrives.
Also, to fix OpenTX bug mentioned in older PR, use old deflection rate if channel value did not change (repeated frame) and handle 'time since last RC update' correctly ...